currencies/rate_converter.go

```diff
@@ -160,14 +168,22 @@ func (rc *RateConverter) LastUpdated() time.Time {
 
 // Rates returns current conversions rates
 func (rc *RateConverter) Rates() Conversions {
-	if rc.constantRates != nil {
-		// Converter is not active, returning the constant rates
-		return rc.constantRates
-	}
-	if rates := rc.rates.Load(); rates != nil {
+	if rates := rc.rates.Load(); rates != nil && rates != (*Rates)(nil) {
 		return rates.(*Rates)
 	}
-	return nil
+	return rc.constantRates
+}
+
+// CheckStaleRates checks if loaded third party conversion rates are stale
+func (rc *RateConverter) CheckStaleRates() bool {
+	currentTime := time.Now().UTC()
```

Comment:
We likely wanna add a check here to make sure the `staleRatesThreshold` value is greater than `0`. If it is `0` then we should never consider the rates stale i.e. the same behavior that we had before.